### PR TITLE
feat: email ich

### PR DESCRIPTION
**Pull request by webdeveloper001**

### CODE REVIEW

File: api/controllers/creator/ich-email.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/**
+ * creator/ich-email.js
```
Comment (tomer-tgp) [inline api/controllers/creator/ich-email.js:2]: let's change the file name to `engagement-email` 
It is a more reusable name, and the backend naming should not be tied to the front end action (imagine if they change that to another verb)

File: api/controllers/creator/ich-email.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            type: 'string',
+            required: true,
+        },
+        projectName: {
+            friendlyName: 'Project Name',
+            description: 'name of project',
+            type: 'string',
+            required: true,
+        },
+        userId: {
```
Comment (tomer-tgp) [inline api/controllers/creator/ich-email.js:49]: Doesn't look like you are using this userId

File: api/controllers/creator/ich-email.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    },
+
+    fn: async function (inputs, exits) {
+        try {
+            const { message, senderEmail, creatorName, creatorEmail, senderName, projectName, userId } = inputs;
+            const subject = `❤️ ${senderName} can help with ${projectName}`;
+            const messageHeader = '';
+            const email = creatorEmail;
+
+            const name = creatorName;
```
Comment (tomer-tgp) [inline api/controllers/creator/ich-email.js:74]: why not just use `creatorName`? it is clearer

File: api/controllers/creator/ich-email.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                messageHeader,
+                messageBody,
+                toEmail,
+                ccEmail
+            );
+            return exits.success({
+                message: 'Email Sent Successfully',
+            });
+        } catch (err) {
+            console.log('email sent error');
```
Comment (tomer-tgp) [inline api/controllers/creator/ich-email.js:106]: please change to `console.log('creators email sent error');` it helps me debug the server when needed.

File: api/helpers/mailgun-sender.js [hunk 5]
```diff
@@ -29,18 +29,29 @@ module.exports = {
...
-      const { message, messageHeader, email, name, subject } = inputs;
+      const { message, messageHeader, email, name, subject, fromEmail, ccEmail } = inputs;
       const MAILGUN_API =
         sails.config.custom.MAILGUN_API || sails.config.MAILGUN_API;
       const mg = mailgun.client({ username: 'api', key: MAILGUN_API });
 
       mg.messages
         .create('mg.thegoodparty.org', {
-          from: 'NoReply@TheGoodParty.org <noreply@thegoodparty.org>',
+          from: fromEmail || 'NoReply@TheGoodParty.org <noreply@thegoodparty.org>',
```
Comment (tomer-tgp) [inline api/helpers/mailgun-sender.js:53]: you can't send emails from other domains. You will get a spam flag. You can only send the emails from our domain

File: api/controllers/creator/ich-email.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        },
+        badRequest: {
+            description: 'Error sending email',
+            responseType: 'badRequest',
+        },
+    },
+
+    fn: async function (inputs, exits) {
+        try {
+            const { message, senderEmail, creatorName, creatorEmail, senderName, projectName, userId } = inputs;
```
Comment (tomer-tgp) [inline api/controllers/creator/ich-email.js:69]: There is a problem here. 
This endpoint is not authenticated, and anyone can use it to send emails to anyone.

File: api/controllers/notifications/email-ama.js [hunk 7]
```diff
@@ -42,7 +42,7 @@ module.exports = {
         name,
         subject,
         messageHeader,
-        msgWithLineBreaks,
+        msgWithLineBreaks
```
Comment (tomer-tgp) [inline api/controllers/notifications/email-ama.js:45]: why?

File: api/controllers/creator/ich-email.js (hunk 6 above)
Comment (tomer-tgp) [reply api/controllers/creator/ich-email.js:69]: If we can ask users to register first, we can take their email from the user

File: api/helpers/mailgun-sender.js (hunk 5 above)
Comment (webdeveloper001) [reply api/helpers/mailgun-sender.js:53]: you mean, we have to keep NoReply@TheGoodParty.org as from email. I was thinking that project creators might be able to reply to the sender's email directly.

File: api/controllers/creator/engagement-email.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
+/**
+ * creator/ich-email.js
+ *
+ * @description :: Sends and email to stakeholders when user submits a form.
+ * @help        :: See https://sailsjs.com/documentation/concepts/actions-and-controllers
+ */
+module.exports = {
+    friendlyName: 'I CAN HELP Email',
```
Comment (tomer-tgp) [inline api/controllers/creator/engagement-email.js:8]: please change to "Engagement Email"

File: api/controllers/creator/engagement-email.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
+/**
+ * creator/ich-email.js
```
Comment (tomer-tgp) [inline api/controllers/creator/engagement-email.js:2]: change the name here too

Also, let's change the directory to "creators" to match the name we are using everywhere.

**Pull request closed without merging**